src/MySqlConnector/MySql.Data.MySqlClient/MySqlDataReader.cs

```diff
@@ -273,6 +273,8 @@ public override long GetChars(int ordinal, long dataOffset, char[] buffer, int b
 #if !NETSTANDARD1_3
 		public override DataTable GetSchemaTable()
 		{
+			if (m_resultSet is null)
+				throw new InvalidOperationException("There is no current result set.");
```

Comment:
`BuildSchemaTable` calls `GetResultSet`, which already performs this check.

I don't know offhand where the test needs to happen, but this isn't the right test.